### PR TITLE
#57: added double literal

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: src/main/java/com/exasol/sql/ValueTableRow.java [hunk 1]
```diff
@@ -108,6 +115,17 @@ public Builder add(final int... values) {
...
         }
 
+        /**
+         * Add one or more double literals to the row.
+         *
+         * @param values doubles to be added
+         * @return <code>this</code> for fluent programming
+         */
+        public Builder add(final double... values) {
+            stream(values).mapToObj(DoubleLiteral::of).forEach(this.expressions::add);
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/ValueTableRow.java]: Please use the same way as shown above.
While Streams are useful, they are slower in this scenario and the code is harder to read.

Comment (redcatbear) [reply src/main/java/com/exasol/sql/ValueTableRow.java]: Sonar Cloud analysis failed for reasons that are not clear. Let's investigate this together.

Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/sql/ValueTableRow.java]: :( okay

File: src/main/java/com/exasol/sql/ValueTable.java [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package com.exasol.sql;
 
-import java.util.*;
+import java.util.ArrayList;
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/ValueTable.java]: I am going to check the formatter settings again. Please correct by hand until then.

In the `org.eclipse.jdt.ui.prefs` I see:

```
org.eclipse.jdt.ui.ondemandthreshold=3
org.eclipse.jdt.ui.staticondemandthreshold=3
```

Both are correct.

That means the problem must be on the IntelliJ side.

File: src/main/java/com/exasol/sql/ValueTable.java [hunk 3]
```diff
@@ -63,6 +65,15 @@ public void add(final String... values) {
         amendLastRow(createLastRowBuilder().add(values).build());
     }
 
+    /**
+     * Add char values to the last row of the value table.
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/ValueTable.java]:
```suggestion
     * Add character values to the last row of the value table.
```

File: src/main/java/com/exasol/sql/ValueTableRow.java [hunk 4]
```diff
@@ -1,6 +1,9 @@
 package com.exasol.sql;
 
-import java.util.*;
+import static java.util.Arrays.asList;
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/ValueTableRow.java]: See above.

File: src/main/java/com/exasol/sql/ValueTableRow.java [hunk 5]
```diff
@@ -19,7 +22,7 @@
      */
     public ValueTableRow(final Fragment root, final ValueExpression... expressions) {
         super(root);
-        this.expressions = Arrays.asList(expressions);
+        this.expressions = asList(expressions);
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/ValueTableRow.java]: Static method access through class should be explicit in this case. Please correct. Looks like IntelliJ is changing that too.

File: src/main/java/com/exasol/sql/ValueTableRow.java [hunk 6]
```diff
@@ -108,6 +124,58 @@ public Builder add(final int... values) {
...
+
+        /**
+         * Add one or more float literals to the row.
+         *
+         * @param values floats to be added
+         * @return <code>this</code> for fluent programming
+         */
+        public Builder add(final float... values) {
+            for (final double value : values) {
+                this.expressions.add(DoubleLiteral.of(value));
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/ValueTableRow.java]:
```suggestion
                this.expressions.add(FloatLiteral.of(value));
```

File: src/main/java/com/exasol/sql/expression/ExpressionTerm.java [hunk 7]
```diff
@@ -18,6 +18,16 @@ public static StringLiteral stringLiteral(final String value) {
         return StringLiteral.of(value);
     }
 
+    /**
+     * Create a string literal from char.
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/expression/ExpressionTerm.java]:
```suggestion
     * Create a string literal from a character.
```

File: src/main/java/com/exasol/sql/expression/ExpressionTerm.java [hunk 8]
```diff
@@ -18,6 +18,16 @@ public static StringLiteral stringLiteral(final String value) {
...
+     *
+     * @param value char value
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/expression/ExpressionTerm.java]:
```suggestion
     * @param value character value
```

File: src/main/java/com/exasol/sql/expression/ExpressionTerm.java [hunk 9]
```diff
@@ -28,6 +38,46 @@ public static IntegerLiteral integerLiteral(final int value) {
...
+     */
+    public static FloatLiteral floatLiteral(final float value) {
+        return FloatLiteral.of(value);
+    }
+
+    /**
+     * Create a boolean literal.
+     *
+     * @param value literal value
+     * @return double literal
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/expression/ExpressionTerm.java]:
```suggestion
     * @return boolean literal
```

File: src/main/java/com/exasol/sql/expression/StringLiteral.java [hunk 10]
```diff
@@ -20,6 +20,16 @@ public static StringLiteral of(final String literal) {
         return new StringLiteral(literal);
     }
 
+    /**
+     * Create a new {@link StringLiteral} from a char.
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/expression/StringLiteral.java]:
```suggestion
     * Create a new {@link StringLiteral} from a character.
```

File: src/main/java/com/exasol/sql/ValueTable.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/sql/ValueTable.java]: fixed

File: src/main/java/com/exasol/sql/ValueTableRow.java (hunk 5 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/sql/ValueTableRow.java]: fixed

File: src/main/java/com/exasol/sql/ValueTableRow.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/sql/ValueTableRow.java]: fixed

**Pull request merged**